cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java

```diff
@@ -72,6 +73,15 @@ public void execute(Project project) {
                 BuildTime.getInstance().packageResolutionDuration = System.currentTimeMillis() - start;
                 start = System.currentTimeMillis();
             }
+
+            // run built-in code generator compiler plugins
+            if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
```

Comment:
This could be a problem for Azure functions/AWS lambda modules. as those modules contain single file projects. I understand from the compiler POV Single File projects can't have more than one sources but from users point of view it can be bit confusing as they are not aware of the internals. 
In this case we might have to turn the single file project into build project if theres additional generated sources. I think this can be very complicated task :/
If we can't fix this issue, i think we have to force users to always use Projects instead of Single file projects when using AWS lambda and Azue functions modules.

@sameerajayasoma  @azinneera

Reply:
I think It is better to restrict your feature only to build projects